### PR TITLE
ci: add a test for running Docker image

### PR DESCRIPTION
**Pull request by komainu8**
Currently, we don't have confirmed whether publish latest docker image or not.
This modification we can confirm automatically whether publish latest docker image or not.

### CODE REVIEW

**Comment by komainu8** [comment]
@kou @HashidaTKS Could you review this PR?

File: .github/workflows/run.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    steps:
+      - name: Run Docker image
+        run: |
+          sudo apt -u -V install \
+            curl \
+            jq
+          latest_version=`curl https://api.github.com/repos/pgroonga/pgroonga/releases/latest | jq .tag_name | sed s/\"//g`
+          tags="${latest_version}-${{ matrix.id }}"
+          docker_image="groonga/pgroonga:${tags}"
+          sudo docker run -d ${docker_image}
```
Comment (HashidaTKS) [inline .github/workflows/run.yml:42]: Do we need  to execute `docker run`?
Is it not enough to check the existence of the tag?

Can we use `https://registry.hub.docker.com/v2/repositories/groonga/pgroonga/tags` ?
If we can use it, how about the following?

```
curl -s https://registry.hub.docker.com/v2/repositories/groonga/pgroonga/tags?page_size=1000 | jq '.results[].name' | grep $latest_version
#and check $? and so on...
```

Comment (komainu8) [reply .github/workflows/run.yml:42]: I think that we should also confirm docker image operate correctly.
Even though we confirm set correct tag, if user can't use this docker image, this docker image is meaningless.

Comment (kou) [reply .github/workflows/run.yml:42]: I like the komainu8's opinion.
How about confirming PGroonga's version too?

File: .github/workflows/run.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          - "debian-14"
+          - "debian-13"
+          - "debian-12"
+    steps:
+      - name: Run Docker image
+        run: |
+          sudo apt -u -V install \
+            curl \
+            jq
+          latest_version=`curl https://api.github.com/repos/pgroonga/pgroonga/releases/latest | jq .tag_name | sed s/\"//g`
```
Comment (kou) [inline .github/workflows/run.yml:39]:
```suggestion
          latest_version=$(curl https://api.github.com/repos/pgroonga/pgroonga/releases/latest | jq -r .tag_name)
```

File: .github/workflows/run.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        id:
+          - "alpine-15"
+          - "alpine-14"
+          - "alpine-13"
+          - "alpine-12"
+          - "alpine-11"
+          - "alpine-10"
```
Comment (kou) [inline .github/workflows/run.yml:22]: PGroonga dropped support for PostgreSQL 10.

File: .github/workflows/run.yml (hunk 1 above)
Comment (HashidaTKS) [reply .github/workflows/run.yml:42]: I see.

**Comment by komainu8** [comment]
Thank you for your comments.
I've fixed them.

File: .github/workflows/run.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          tags="${latest_version}-${{ matrix.id }}"
+          docker_image="groonga/pgroonga:${tags}"
+          sudo docker run \
+            -p 5432:5432 \
+            -e POSTGRES_DB=pgroonga \
+            -e POSTGRES_PASSWORD=pgroonga \
+            -e POSTGRES_USER=pgroonga \
+            -d ${docker_image}
+          for i in {1..60}; do
+            if pg_isready -h 127.0.0.1 ; then
```
Comment (kou) [inline .github/workflows/run.yml:51]:
```suggestion
            if pg_isready -h 127.0.0.1; then
```

File: .github/workflows/run.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          psql \
+            -h 127.0.0.1 \
+            -U pgroonga \
+            -c "CREATE EXTENSION pgroonga;"
+          pgroonga_version=`psql \
+                             -h 127.0.0.1 \
+                             -U pgroonga \
+                             -t \
+                             -A \
+                             -c "SELECT extversion FROM pg_catalog.pg_extension WHERE extname = 'pgroonga';"`
```
Comment (kou) [inline .github/workflows/run.yml:68]: Could you use `$(...)` for capturing command output?

File: .github/workflows/run.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+          if [ "${latest_version}" != "${pgroonga_version}" ]; then
+            exit 1
+          fi
```
Comment (kou) [inline .github/workflows/run.yml:71]:
```suggestion
          set -x
          [ "${latest_version}" == "${pgroonga_version}" ]
```

File: .github/workflows/run.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+              break
+            fi
+            sleep 1
+          done
+          if [ $i -eq 60 ]; then
+            echo "PostgreSQL not avalable"
+          fi
+          psql \
+            -h 127.0.0.1 \
+            -U pgroonga \
```
Comment (kou) [inline .github/workflows/run.yml:61]: Why can we connect without password?

Comment (komainu8) [reply .github/workflows/run.yml:61]: I set "PGPASSWORD" to environment value in 
https://github.com/pgroonga/docker/pull/18/files#diff-a3ddd7238fc6e36daeb0ef76e93fe15cc87824bd3299888075210c5ca6a474d3R33.

Comment (kou) [reply .github/workflows/run.yml:61]: Ah, I see.

File: .github/workflows/run.yml [hunk 8]
```diff
@@ -0,0 +1,71 @@
+name: Run
```
Comment (kou) [inline .github/workflows/run.yml:1]: How about "Test"?

File: .github/workflows/run.yml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+on:
+  schedule:
+    - cron: |
+        0 0 * * *
+  pull_request:
+concurrency:
+  group: ${{ github.head_ref || github.sha }}-${{ github.workflow }}
```
Comment (kou) [inline .github/workflows/run.yml:8]: We can use literal here because this is not related to a commit. This is only ran periodically.

```suggestion
  group: test
```

File: .github/workflows/run.yml [hunk 10]
```diff
@@ -0,0 +1,71 @@
+name: Run
+on:
+  schedule:
+    - cron: |
+        0 0 * * *
+  pull_request:
```
Comment (kou) [inline .github/workflows/run.yml:6]: We should remove this on merge, right?

**Comment by kou** [comment]
How about solving #17 and pushing tags before we merge this?

File: .github/workflows/run.yml (hunk 10 above)
Comment (komainu8) [reply .github/workflows/run.yml]: Yes.
I'll remove this definition before merge.
Currently, I remain this definition for debugging.

File: .github/workflows/run.yml (hunk 9 above)
Comment (komainu8) [reply .github/workflows/run.yml]: Oh. I see.

**Comment by komainu8** [comment]
> How about solving https://github.com/pgroonga/docker/issues/17 and pushing tags before we merge this?

OK.
I solve https://github.com/pgroonga/docker/issues/17 before this PR.

**Comment by komainu8** [comment]
I've rebased master.
This PR's CI passed all tests with the latest tag. See https://github.com/pgroonga/docker/actions/runs/3727658848/jobs/6322001149.

File: .github/workflows/run.yml (hunk 10 above)
Comment (komainu8) [reply .github/workflows/run.yml]: I've removed this in https://github.com/pgroonga/docker/pull/18/commits/48eadf308dd73217c443bba67a1875fc7c3793b2.
